microcosm/loaders/compose.py

```diff
@@ -14,6 +16,37 @@ def merge(configs):
     return result
 
 
+def two_stage_loader(
+    primary_loader: Loader,
+    secondary_loader: SecondaryLoader,
+    prefer_secondary: bool = False,
+) -> Loader:
+    """
+    Returns a loader that will first call the `initial_loader`, then call
+    `secondary_loader`, passing it the output of `initial_loader`.  It merges
+    the output of the two functions.  In the case where the config output of
+    the two functions have overlapping keys, the `prefer_secondary` parameter
+    determines whether the secondary or primary output takes precendence.
+
+    """
+    def loader(metadata: Metadata) -> Configuration:
+        primary_config = Configuration(primary_loader(metadata))
+        secondary_config = secondary_loader(metadata, primary_config)
+
+        if prefer_secondary:
+            return merge([
+                primary_config,
+                secondary_config,
+            ])
+        else:
+            return merge([
+                secondary_config,
+                primary_config,
+            ])
+
+    return loader
+
+
```

Comment:
While working on something else, I found one more use case for this function, and now all 3 of my use cases have only two stages, and merge the result of the second stage in with the primary config, letting the primary config take precedence.  Thus I refactored to specifically target that use case.  If we end up needing more generality we can refactor.  Make sense?